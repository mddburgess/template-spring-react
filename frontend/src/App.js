import React, {useState, useEffect} from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
    const [message, setMessage] = useState(null);

    async function fetchMessage() {
        const response = await fetch('api/hello');
        setMessage(await response.text());
    }

    useEffect(() => {
        fetchMessage();
    });

    return (
        <div className="App">
            <header className="App-header">
                <img src={logo} className="App-logo" alt="logo"/>
                <p>{message}</p>
                <a
                    className="App-link"
                    href="https://reactjs.org"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    Learn React
                </a>
            </header>
        </div>
    );
}

export default App;
