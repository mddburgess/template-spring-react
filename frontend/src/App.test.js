import React from 'react';
import {act, render} from '@testing-library/react';
import App from './App';

test('renders', async () => {
    fetch.mockResponse('Hello Test!');

    await act(async () => {
        await render(<App/>);
    });

    expect(document.body).toMatchSnapshot();
});
