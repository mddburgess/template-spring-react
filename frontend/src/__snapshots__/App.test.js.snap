// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <img
          alt="logo"
          class="App-logo"
          src="logo.svg"
        />
        <p>
          Hello Test!
        </p>
        <a
          class="App-link"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React
        </a>
      </header>
    </div>
  </div>
</body>
`;
